### PR TITLE
feat: ✨ merge build tool config by ast

### LINKED ISSUE #36: 调研并实现插件的基本结构，给出更具体的开发方案、构建工具适配方案
插件的基本结构如下：
```
plugin-xxx
├── README.md
├── generator
│   └── template # 相关文件
│   └── index.js  # generator
├── prompts.js    # 命令行提示安装
├── index.js      # service 插件
└── package.json
```

### PR DESCRIPTION
**Pull request by haoyanwang**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/xun082/create-neat/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[✅] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

- Closes #36 
- Related to #24  

## What is the new behavior?
使用unplugin适配不同构建工具

## Does this PR introduce a breaking change?

```
[✅] Yes
[ ] No
```

### CODE REVIEW

**Comment by uaenaTzx** [comment]
Related to 要关联到父issue

File: packages/core/package.json [hunk 1]
```diff
@@ -43,7 +43,12 @@
     "fs-extra": "^11.2.0",
     "minimist": "^1.2.8",
     "ora": "^5.4.1",
-    "tar": "^6.2.0"
+    "tar": "^6.2.0",
+    "webpack-chain": "^6.5.1",
+    "@babel/parser": "^7.24.0",
```
Comment (uaenaTzx) [inline packages/core/package.json:48]: 为什么这些会加在 core 里面

File: packages/core/src/utils/Generator.ts [hunk 2]
```diff
@@ -109,6 +114,24 @@ class Generator {
         this.plugins[pluginName],
         this.rootOptions,
       );
+      // 执行plugin的入口文件，把config写进来
+      const pluginEntry = await loadModule(
+        `packages/@plugin/plugin-${pluginName.toLowerCase()}/index.cjs`,
+        path.resolve(__dirname, relativePathToRoot),
+      );
+      // 处理webpack config
+      if (this.buildToolConfig.buildTool === "webpack") {
```
Comment (uaenaTzx) [inline packages/core/src/utils/Generator.ts:123]: 这一步逻辑能抽象出来吗

File: packages/@plugin/plugin-babel/index.cjs [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const buildToolConfigs = {
+  // 支持拓展loader和plugin
+  webpack: () => {
+    const a = new RegExp();
```
Comment (qq1051766345) [inline packages/@plugin/plugin-babel/index.cjs:4]: 这个正则是不是没用上，没用上可以删了

Comment (haoyanwang) [reply packages/@plugin/plugin-babel/index.cjs:4]: 嗯，测试代码，忘记删了

File: packages/core/src/utils/Generator.ts (hunk 2 above)
Comment (uaenaTzx) [reply packages/core/src/utils/Generator.ts:123]: 我是想可以考虑把构建工具的处理独立一个文件出来，先这样吧，我后面整理一下

File: packages/core/package.json (hunk 1 above)
Comment (haoyanwang) [reply packages/core/package.json:48]: 之前尝试用webpack-chain，后来发现不需要了，删掉

File: packages/core/src/utils/ast.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import {
```
Comment (uaenaTzx) [inline packages/core/src/utils/ast.ts:1]: 这个文件一些相关函数尽量给一些 jsdoc 注释

Comment (haoyanwang) [reply packages/core/src/utils/ast.ts:1]: done

**Review by uaenaTzx — APPROVED** [review]

**Pull request merged**